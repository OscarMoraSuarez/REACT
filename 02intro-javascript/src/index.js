const loguearMensaje = () => {

    let activo = true;
    let mensaje;

   /*  mensaje = (activo) ? 'usuario activo' : 'usuario inactivo'; */
   mensaje=activo&&'activo';
    return mensaje;

}

console.log(loguearMensaje());