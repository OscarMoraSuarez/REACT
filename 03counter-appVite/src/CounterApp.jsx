import { useState }from 'react';
import { PropTypes } from "prop-types";

//?cunado la funcion no ocupa nada en el componente en el cual se aloja
//? se recomienda declararla y usarla afuera



const CounterApp=({value})=>{

    const [ counter,setCounter ]=useState(value)
    
    const sumar = () => {

        // setCounter(counter+1);
        setCounter(counter+1)

    }
     const restar = () =>  setCounter(counter - 1);

    
    const resetear = () => setCounter( value) ;

    
   

return(

    <>
        <h1> CounterApp </h1>
        <h2 data-testid="counter"> { counter } </h2>
        <button aria-label='btn-sumar' onClick={ sumar }> 
        +1
        </button>
        <button aria-label='btn-restar' onClick={ restar }> 
        -1 
        </button >
        <button id='btn-reset' onClick={ resetear }> 
        reset
        </button>
    </>

)
    
    




}

export {

    CounterApp

}


CounterApp.propTypes={

    value:PropTypes.number.isRequired


}

/* CounterApp.defaultProps={

    value:100, 

}
 */
    