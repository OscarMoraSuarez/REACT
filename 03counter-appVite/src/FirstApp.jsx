import PropTypes from 'prop-types';


/* const saludar=(nombre)=>{

    return nombre.toUpperCase();

} */

const FirstApp = ({title,subtitle,name }) => {



return (
    <>      
          <h1>{title}</h1>
          <p>{subtitle}</p>
          <span>{name}</span>
    </>
    )
}

export{

    FirstApp

}

FirstApp.propTypes={
    title:PropTypes.string.isRequired,
    subtitle: PropTypes.number.isRequired,



}

/* estas propiedades entran antes que lso PropTypes */
FirstApp.defaultProps={
    name:'RaJSo',
    subtitle:'123',
    title:'No hay titulo',

     

}