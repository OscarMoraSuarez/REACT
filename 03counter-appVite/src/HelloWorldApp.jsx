// function App() {

//     return <h1>Hola Mundo!!!</h1>;

// }
// export default App;


export const HelloWorld = () => {
  return (
    <h1>HelloWorld App</h1>
  )
}
