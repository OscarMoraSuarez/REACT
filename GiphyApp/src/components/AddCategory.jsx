import { useState } from "react"
import PropTypes from "prop-types"
 

export const AddCategory = ( {onNewCategory}) => {
    const [inputValue, setInputValue] = useState('');
    //capturo el valor del input
    const onInputChange=({target})=>{
        
        setInputValue(target.value);



    }
    // capturo el evento Submit
    const onSubmit=(event)=>{
        
        const textValue=inputValue.trim();
        event.preventDefault();
        if(textValue.length<=1) return;
        // setCategories(categories=>[inputValue,...categories]);
        setInputValue('');
        onNewCategory(textValue);
        

    }

  return (
    
        <form onSubmit={onSubmit} aria-label='form'>
              <input
                  type="text"
                  placeholder="seek Gifs"
                  value={inputValue}
                  onChange={onInputChange}
                />
        </form>
    )
}
AddCategory.propTypes={

    onNewCategory:PropTypes.func.isRequired


}