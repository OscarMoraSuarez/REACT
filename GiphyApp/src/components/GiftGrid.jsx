import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from "prop-types";

export const GiftGrid = ({category}) => {
    
    const {images,isLoading}=useFetchGifs(category);
    console.log({isLoading});
    
    return (
      <>
          <h3>{category}</h3>
          {
            isLoading && (<h2>Cargando...</h2>)
          }

          <div className='card-grid'>
            
          {images.map(( img ) =>(
            <GifItem 
              key={img.id} 
              {...img}
              
            /> ))
          } 

        </div>
      </>
    )
}
GiftGrid.propTypes={

  category:PropTypes.string.isRequired

}
