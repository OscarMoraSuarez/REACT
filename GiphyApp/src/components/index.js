
export * from './AddCategory';
export * from './GiftGrid';
export * from './GifItem';
/* export * from './Saludo'; */