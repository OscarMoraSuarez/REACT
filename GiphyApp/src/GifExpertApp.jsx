import { useState } from "react";
import { AddCategory,GiftGrid} from "./components";





export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch']);
    const onAddCategory=(NewCategory)=>{
      if (categories.includes(NewCategory)) return;
         /* categories.push(NewCategory);  */
         setCategories([...categories,NewCategory]); 
        /* setCategories(NewCategory=>[...categories,NewCategory]); */
        console.log(NewCategory);
    }


  return (
    <>  
        <h1>GifExpertApp</h1>
          <AddCategory 
          /* setCategories={setCategories}  */
          onNewCategory={value=>onAddCategory(value)}
          />
        
        
        {categories.map(category => (
                <GiftGrid 
                  key={category} 
                  category={category}
                />
            )
          )
        }
        

        
    </>
  )
}
