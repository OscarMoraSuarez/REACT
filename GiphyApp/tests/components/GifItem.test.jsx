import {  render, screen } from "@testing-library/react"
import { checkPropTypes } from "prop-types"
import { GifItem } from "../../src/components"


describe('pruebas en <GifItem/>', () => {  
     
    const url='https://cualquiercosa/';
    const title='raJSo'
    test('debe hacer juego con el snapshot', () => { 

        const{ container } = render(<GifItem url={url} title={title} />)
        

        expect(container).toMatchSnapshot();

     })

     test('debe msotrar la imagen con  el url u el title indicado', () => { 
        
         render(<GifItem url={url} title={title}/>)
        //  screen.debug();
        //  expect(screen.getByRole('img').src).toBe(url);
        //  expect(screen.getByRole('img').alt).toBe(title);
        const {src,alt}=screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);

    })
    test('Debe mostrar el titulo en el componente', () => { 
        render(<GifItem url={url} title={title} />)
        expect(screen.getByText(title)).toBeTruthy();

     })


})