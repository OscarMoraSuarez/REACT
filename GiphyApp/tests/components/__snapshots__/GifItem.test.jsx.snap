// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifItem/> debe hacer juego con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="raJSo"
      src="https://cualquiercosa/"
    />
    <p>
      raJSo
    </p>
  </div>
</div>
`;
